File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼.py
def factorial(num):
    if num == 1:
        return 1
    return num * factorial(num - 1)

def solution(n):
    answer = 0
    i = 1
    while True:
        if n >= factorial(i):
            i += 1
        else:
            return i - 1
    return answer

File: 백준/Bronze/10872. 팩토리얼/팩토리얼.js
const fs = require("fs");
const readFileSyncAddress = '/dev/stdin';

let input = fs.readFileSync(readFileSyncAddress).toString().trim()

let value = Number(input)

function solution(num) {
  if (num === 1 || num === 0) return 1
  return num * solution(num - 1)
}

console.log(solution(value))